Source/Core/AudioCommon/XAudio2_7Stream.cpp

```diff
@@ -58,21 +72,38 @@ void StreamingVoiceContext2_7::SubmitBuffer(PBYTE buf_data)
 
 StreamingVoiceContext2_7::StreamingVoiceContext2_7(IXAudio2* pXAudio2, Mixer* pMixer,
                                                    Common::Event& pSyncEvent)
-    : m_mixer(pMixer), m_sound_sync_event(pSyncEvent),
-      xaudio_buffer(new BYTE[NUM_BUFFERS * BUFFER_SIZE_BYTES]())
+    : m_mixer(pMixer), m_sound_sync_event(pSyncEvent)
 {
   WAVEFORMATEXTENSIBLE wfx = {};
-
   wfx.Format.wFormatTag = WAVE_FORMAT_EXTENSIBLE;
   wfx.Format.nSamplesPerSec = m_mixer->GetSampleRate();
-  wfx.Format.nChannels = 2;
-  wfx.Format.wBitsPerSample = 16;
+
+  if (m_use_surround)
+  {
+    xaudio_buffer = std::unique_ptr<BYTE[]>(new BYTE[NUM_BUFFERS * BUFFER_SIZE_BYTES_SURROUND]);
```

Comment:
this can be simplified to:

```cpp
xaudio_buffer = std::unique_ptr<BYTE[]>(NUM_BUFFERS * BUFFER_SIZE_BYTES_SURROUND);
```

ditto for the other one.

Reply:
I tried changing to that but I couldn't, it won't compile.

IntelliSense error:
```
Error (active) E0289 no instance of constructor "std::unique_ptr<_Ty [], _Dx>::unique_ptr [with _Ty=BYTE, _Dx=std::default_delete<BYTE []>]" matches the argument list
```

Reply:
Oh sorry, I had meant to type `std::make_unique`, not `std::unique_ptr`. So:

```cpp
xaudio_buffer = std::make_unique<BYTE[]>(NUM_BUFFERS * BUFFER_SIZE_BYTES_SURROUND);
```